core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -924,6 +924,33 @@ class AuthorizerIntegrationTest extends AbstractAuthorizerIntegrationTest {
     sendRequests(requestKeyToRequest, false, topicNames)
   }
 
+  @Test
+  def testAuthorizationWithTopicNotExistingForProduceReqeustVersionLessThan13(): Unit = {
+    for (version <- ApiKeys.PRODUCE.oldestVersion to 12) {
+      val request = requests.ProduceRequest.builder(new ProduceRequestData()
+          .setTopicData(new ProduceRequestData.TopicProduceDataCollection(
+            util.List.of(new ProduceRequestData.TopicProduceData()
+                .setName(tp.topic())
```

Comment:
This topic exists on the broker, right?

Reply:
We expect to return `TOPIC_AUTHORIZATION_FAILED` (if it failed to authorize) regardless of topic existence if the request is using topic name. Maybe we should test both cases - 1) topic exists and 2) topic does not exist (similar to `testAuthorizationFetchV12WithTopicNotExisting`)

Reply:
We could. I am just pointing out this doesn't match the test name, which says `WithTopicNotExisting`.

Reply:
Hi @junrao, This case follows `testAuthorizationWithTopicNotExisting`. In `AuthorizerIntegrationTest`, if a case needs a existing topic like case `testAuthorizationWithTopicExisting`, it needs to call `sendRequests(mutable.Map(ApiKeys.CREATE_TOPICS -> createTopicsRequest))`.

Reply:
@FrankYang0529 could you please add the test case for the existent topic? Additionally, it would be useful to add case of existent topic for fetch RPC too.